job_runner/reporting/event.py

```diff
@@ -87,78 +98,58 @@ def _init_handler(self, msg):
         self._step_id = msg.step_id
         self._event_publisher_thread.start()
 
-    def _job_handler(self, msg):
-        job_path = f"{self._step_path()}/job/{msg.job.index}"
-
+    def _job_handler(self, msg: Message):
+        job_name = msg.job.name()
+        logger.debug(f"Job: {job_name} is being handle")
+        job_data = {
+            "source": f"{self._step_path()}/job/{msg.job.index}",
+            "datacontenttype": "application/json",
```

Comment:
But the `data` field varies. Sometimes there's JSON, sometimes it is `None`. From the CloudEvent spec:

> if the `datacontenttype` declares the data to contain JSON-formatted content, a JSON serializer MUST translate the data value to a JSON value, and use the member name data to store it inside the JSON representation. 

Even though omitting the `datacontettype` defaults to JSON if the format of the CloudEvent itself is JSON, I think it's best to omit `datacontettype` when `data` is `None`.

Reply:
I added the following to `_dump_cloud_event`
```python
if data is None and _CONTENT_TYPE in attributes:
    attributes.pop(_CONTENT_TYPE)
```